Saved credentials encryption is DISABLED for this Metabase instance.

### Bugs
When   I run lein ring server,
will show this:
Compiling metabase.logger
Compiling metabase.task.DynamicClassLoadHelper
08-21 10:27:10 INFO metabase.util :: Loading Metabase...
08-21 10:27:17 INFO util.encryption :: Saved credentials encryption is DISABLED for this Metabase instance. 🔓 
For more information, see https://www.metabase.com/docs/latest/operations-guide/start.html#encrypting-your-database-connection-details-at-rest
08-21 10:28:18 INFO metabase.core :: Starting Metabase version v0.30.1 (null null) ...
08-21 10:28:18 INFO metabase.core :: System timezone is 'Asia/Shanghai' ...
Exception in thread "main" java.lang.NullPointerException, compiling:(google.clj:51:3)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3700)
	at clojure.lang.Compiler$DefExpr.eval(Compiler.java:457)
	at clojure.lang.Compiler.eval(Compiler.java:7067)
	at clojure.lang.Compiler.load(Compiler.java:7514)
	at clojure.lang.RT.loadResourceScript(RT.java:379)
	at clojure.lang.RT.loadResourceScript(RT.java:370)
	at clojure.lang.RT.load(RT.java:460)
	at clojure.lang.RT.load(RT.java:426)
	at clojure.core$load$fn__6548.invoke(core.clj:6046)
	at clojure.core$load.invokeStatic(core.clj:6045)
	at clojure.core$load.doInvoke(core.clj:6029)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invokeStatic(core.clj:5848)
	at clojure.core$load_one.invoke(core.clj:5843)
	at clojure.core$load_lib$fn__6493.invoke(core.clj:5888)
	at clojure.core$load_lib.invokeStatic(core.clj:5887)
	at clojure.core$load_lib.doInvoke(core.clj:5868)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$load_libs.invokeStatic(core.clj:5929)
	at clojure.core$load_libs.doInvoke(core.clj:5909)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$require.invokeStatic(core.clj:5947)
	at clojure.core$require.doInvoke(core.clj:5947)
	at clojure.lang.RestFn.invoke(RestFn.java:1289)
	at metabase.driver.bigquery$eval65222$loading__6434__auto____65223.invoke(bigquery.clj:1)
	at metabase.driver.bigquery$eval65222.invokeStatic(bigquery.clj:1)
	at metabase.driver.bigquery$eval65222.invoke(bigquery.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:7062)
	at clojure.lang.Compiler.eval(Compiler.java:7051)
	at clojure.lang.Compiler.load(Compiler.java:7514)
	at clojure.lang.RT.loadResourceScript(RT.java:379)
	at clojure.lang.RT.loadResourceScript(RT.java:370)
	at clojure.lang.RT.load(RT.java:460)
	at clojure.lang.RT.load(RT.java:426)
	at clojure.core$load$fn__6548.invoke(core.clj:6046)
	at clojure.core$load.invokeStatic(core.clj:6045)
	at clojure.core$load.doInvoke(core.clj:6029)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invokeStatic(core.clj:5848)
	at clojure.core$load_one.invoke(core.clj:5843)
	at clojure.core$load_lib$fn__6493.invoke(core.clj:5888)
	at clojure.core$load_lib.invokeStatic(core.clj:5887)
	at clojure.core$load_lib.doInvoke(core.clj:5868)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$load_libs.invokeStatic(core.clj:5925)
	at clojure.core$load_libs.doInvoke(core.clj:5909)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$require.invokeStatic(core.clj:5947)
	at clojure.core$require.doInvoke(core.clj:5947)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at metabase.driver$init_driver_in_namespace_BANG_.invokeStatic(driver.clj:323)
	at metabase.driver$init_driver_in_namespace_BANG_.invoke(driver.clj:322)
	at metabase.driver$find_and_load_drivers_BANG_.invokeStatic(driver.clj:334)
	at metabase.driver$find_and_load_drivers_BANG_.invoke(driver.clj:328)
	at metabase.core$init_BANG_.invokeStatic(core.clj:156)
	at metabase.core$init_BANG_.invoke(core.clj:138)
	at clojure.lang.Var.invoke(Var.java:377)
	at ring.server.standalone$serve.invokeStatic(standalone.clj:96)
	at ring.server.standalone$serve.doInvoke(standalone.clj:75)
	at clojure.lang.RestFn.invoke(RestFn.java:423)
	at ring.server.leiningen$serve.invokeStatic(leiningen.clj:19)
	at ring.server.leiningen$serve.invoke(leiningen.clj:16)
	at user$eval65026.invokeStatic(form-init2588511176082585470.clj:1)
	at user$eval65026.invoke(form-init2588511176082585470.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:7062)
	at clojure.lang.Compiler.eval(Compiler.java:7052)
	at clojure.lang.Compiler.load(Compiler.java:7514)
	at clojure.lang.Compiler.loadFile(Compiler.java:7452)
	at clojure.main$load_script.invokeStatic(main.clj:278)
	at clojure.main$init_opt.invokeStatic(main.clj:280)
	at clojure.main$init_opt.invoke(main.clj:280)
	at clojure.main$initialize.invokeStatic(main.clj:311)
	at clojure.main$null_opt.invokeStatic(main.clj:345)
	at clojure.main$null_opt.invoke(main.clj:342)
	at clojure.main$main.invokeStatic(main.clj:424)
	at clojure.main$main.doInvoke(main.clj:387)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.lang.Var.applyTo(Var.java:702)
	at clojure.main.main(main.java:37)
Caused by: java.lang.NullPointerException
	at metabase.driver.google$fn__65244.invokeStatic(google.clj:52)
	at metabase.driver.google$fn__65244.invoke(google.clj:51)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3695)
	... 82 more
08-21 10:28:19 INFO metabase.core :: Metabase Shutting Down ...
08-21 10:28:19 INFO metabase.core :: Metabase Shutdown COMPLETE
Subprocess failed


browser and the version: (Chrome 52.1)
operating system: ( OS X 10)
databases: (sample)
 version: ( 0.30.1)
internal database: (e.x. H2 (default),)

I don't know what caused it. It was possible to run successfully before.

--- a/src/metabase/config.clj
+++ b/src/metabase/config.clj
@@ -61,7 +61,10 @@
 (defn- version-info-from-shell-script []
   (try
     (let [[tag hash branch date] (-> (shell/sh "./bin/version") :out s/trim (s/split #" "))]
-      {:tag tag, :hash hash, :branch branch, :date date})
+      {:tag    (or tag "?")
+       :hash   (or hash "?")
+       :branch (or branch "?")
+       :date   (or date "?")})
     ;; if ./bin/version fails (e.g., if we are developing on Windows) just return something so the whole thing doesn't barf
     (catch Throwable _
       {:tag "?", :hash "?", :branch "?", :date "?"})))

--- a/src/metabase/driver/google.clj
+++ b/src/metabase/driver/google.clj
@@ -46,11 +46,18 @@
   (u/auto-retry 2
     (execute-no-auto-retry request)))
 
+(defn- create-application-name
+  "Creates the application name string, separated out from the `def` below so it's testable with different values"
+  [{:keys [tag ^String hash branch]}]
+  (let [encoded-hash (some-> hash (.getBytes "UTF-8") codec/base64-encode)]
+    (format "Metabase/%s (GPN:Metabse; %s %s)"
+            (or tag "?")
+            (or encoded-hash "?")
+            (or branch "?"))))
+
 (def ^:const ^String application-name
   "The application name we should use for Google drivers. Requested by Google themselves -- see #2627"
-  (let [{:keys [tag ^String hash branch]} config/mb-version-info
-        encoded-hash                      (-> hash (.getBytes "UTF-8") codec/base64-encode)]
-    (format "Metabase/%s (GPN:Metabse; %s %s)" tag encoded-hash branch)))
+  (create-application-name config/mb-version-info))
 
 
 (defn- fetch-access-and-refresh-tokens* [scopes, ^String client-id, ^String client-secret, ^String auth-code]

--- a/test/metabase/driver/google_test.clj
+++ b/test/metabase/driver/google_test.clj
@@ -1,0 +1,26 @@
+(ns metabase.driver.google-test
+  (:require [expectations :refer :all]
+            [metabase.config :as config]
+            [metabase.driver.google :as google]))
+
+;; Typical scenario, all config information included
+(expect
+  "Metabase/v0.30.0-snapshot (GPN:Metabse; NWNjNWY0Mw== master)"
+  (#'google/create-application-name  {:tag "v0.30.0-snapshot", :hash "5cc5f43", :branch "master", :date "2018-08-21"}))
+
+;; It's possible to have two hashes come back from our script. Sending a string with a newline in it for the
+;; application name will cause Google connections to fail
+(expect
+  "Metabase/v0.30.0-snapshot (GPN:Metabse; NWNjNWY0MwphYmNkZWYx master)"
+  (#'google/create-application-name {:tag "v0.30.0-snapshot", :hash "5cc5f43\nabcdef1", :branch "master", :date "2018-08-21"}))
+
+;; It's possible to have all ? values if there was some failure in reading version information, or if non was available
+(expect
+  "Metabase/? (GPN:Metabse; Pw== ?)"
+  (#'google/create-application-name {:tag "?", :hash "?", :branch "?", :date "?"}))
+
+;; This shouldn't be possible now that config/mb-version-info always returns a value, but testing an empty map just in
+;; case
+(expect
+  "Metabase/? (GPN:Metabse; ? ?)"
+  (#'google/create-application-name {}))

